qutebrowser/mainwindow/mainwindow.py

```diff
@@ -561,6 +562,15 @@ def _do_close(self):
             pass
         objreg.get('session-manager').save_last_window_session()
         self._save_geometry()
+        if (
+                len(objreg.window_registry) > 1
+                and all(window is self and self.private or not window.private
```

Comment:
I found this a bit hard to follow. Couldn't you just check `self.private and len(window for window in objreg.window_registry.values() if window.private) == 1` or so?

Reply:
My idea here was to include an extra case: if I'm closing the last private window **but** it is also the last window of any kind, then don't run the cleanup routine. It's just a small improvement to not run unnecesary stuff when shutting down.

Doing `len` instead of `all` would end up in something like:
```
self.private 
and len(objreg.window_registry) > 1
and len(window for window in objreg.window_registry.values() if window.private) == 1
```

which isn't really much shorter, but if you find this clearer I have no problem changing it.

Reply:
Yeah, I was mainly confused by the `all(window is self and self.private or not window.private ...)` part. Maybe adding an extra set of parentheses would make it a bit clearer, but I would indeed prefer the `len(...)` variant (unless you disagree and find the `all(...)` one much clearer).

Maybe also add a comment like `# Wipe private data if we close the last private window, but there are still other windows` or so - I'm usually not a fan of comments duplicating the code's logic, but here I think it'd be worth it.